x-pack/plugins/uptime/public/components/monitor_management/monitor_list/monitor_enabled.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiSwitch } from '@elastic/eui';
+import { EuiSwitchEvent } from '@elastic/eui/src/components/form/switch/switch';
+import { i18n } from '@kbn/i18n';
+import React, { useEffect, useState } from 'react';
+import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
+import { FETCH_STATUS, useFetcher } from '../../../../../observability/public';
+import { ConfigKey, SyntheticsMonitor } from '../../../../common/runtime_types';
+import { setMonitor } from '../../../state/api';
+
+interface Props {
+  id: string;
+  monitor: SyntheticsMonitor;
+  setRefresh: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const MonitorEnabled = ({ id, monitor, setRefresh }: Props) => {
+  const [isEnabled, setIsEnabled] = useState<boolean | null>(null);
+
+  const { notifications } = useKibana();
+
+  const { status } = useFetcher(() => {
+    if (isEnabled !== null) {
+      return setMonitor({ id, monitor: { ...monitor, [ConfigKey.ENABLED]: isEnabled } });
+    }
+  }, [isEnabled]);
+
+  useEffect(() => {
+    if (status === FETCH_STATUS.FAILURE) {
+      notifications.toasts.danger({
+        title: (
+          <p data-test-subj="uptimeMonitorEnabledUpdateFailure">
+            {getMonitorEnabledUpdateFailureMessage(monitor[ConfigKey.NAME])}
+          </p>
+        ),
+        toastLifeTimeMs: 3000,
+      });
+      setIsEnabled(null);
+    } else if (status === FETCH_STATUS.SUCCESS) {
+      notifications.toasts.success({
+        title: (
+          <p data-test-subj="uptimeMonitorEnabledUpdateSuccess">
+            {isEnabled
+              ? getMonitorEnabledSuccessLabel(monitor[ConfigKey.NAME])
+              : getMonitorDisabledSuccessLabel(monitor[ConfigKey.NAME])}
+          </p>
+        ),
+        toastLifeTimeMs: 3000,
+      });
+      setRefresh(true);
+    }
+  }, [status]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Because here I want the side effect to execute only when `status` changes. Without suppressing the linter, it would complain until `isEnabled` is also provided to deps, along with others, which is not desired here.

Reply:
Thanks for the explanation. Could you add an inline comment for context?